spec/spec_helper.rb

```diff
@@ -42,20 +42,20 @@ def set_node_version(version)
   end
 end
 
+def resolve_binary_path
+  (/darwin/ =~ RUBY_PLATFORM) != nil ? './vendor/resolve-version-darwin' : './vendor/resolve-version-linux'
+end
+
 def resolve_node_version(requirements, options = {})
-  # use nodebin to get latest node versions
   requirements.map do |requirement|
-    retry_limit = options[:retry_limit] || 50
-    body = Excon.get("https://nodebin.herokai.com/v1/node/linux-x64/latest?range=#{requirement}", :idempotent => true, :expects => 200, :retry_limit => retry_limit).body
-    JSON.parse(body)['number']
+    result = `#{resolve_binary_path} node #{requirement}`
+    result.split(' ')[0]
   end
 end
 
 def resolve_all_supported_node_versions(options = {})
-  retry_limit = options[:retry_limit] || 50 
-  body = Excon.get("https://nodebin.herokai.com/v1/node/linux-x64/", :idempotent => true, :expects => 200, :retry_limit => retry_limit).body
-  list = JSON.parse(body).map { |n| n['number'] }
-
+  result = `#{resolve_binary_path} list node`
```

Comment:
We've got a `run!` function you can use that will also blow up if a non zero exit code is returned. it's best to use that whenever possible. Otherwise it's like running a bash script without `-e`, it's hard to trace down the actual failure location.

```
   result = run!("#{resolve_binary_path} list node") 
```

Since you've not got that on the node buildpack here's an easy version

```
def run!(cmd)
  out = `#{cmd}`
  raise "Error running command #{cmd.inspect}: #{out}" unless $.success?
  out
end
```